lib/dbutils.js

```diff
@@ -0,0 +1,832 @@
+                : elem.order;
+            orderTerms.push(dbu.fieldName(elem.attribute) + ' ' + dir);
+        }
+    });
+    if (orderTerms.length) {
+        return ' order by ' + orderTerms.join(',') + ' ';
+    } else {
+        return '';
+    }
+}
+
+function constructGroupingByHashKeys(schema) {
+    var groupKeys = [];
+    schema.iKeys.forEach(function(key) {
+        if (schema.iKeyMap[key].type === 'hash') {
+            groupKeys.push(key);
+        }
+    });
+    return ' group by ' + groupKeys.map(dbu.fieldName).join(', ') + ' ';
+}
+
+function constructProj(query, schema) {
+    var projArr = query.proj || schema.proj;
+    var proj;
+    if (Array.isArray(projArr)) {
+        proj = projArr.map(dbu.fieldName).join(',');
+    } else if (projArr.constructor === String) {
+        proj = dbu.fieldName(projArr);
+    } else {
+        throw new Error('Unsupported query proj: ' + projArr + ' of type ' + projArr.constructor);
+    }
+    if (query.distinct) {
+        proj = ' distinct ' + proj + ' ';
+    }
+    return proj;
+}
+
+function isStaticJoinNeeded(query, schema, table) {
+    if (query && query.proj) {
+        if (Array.isArray(query.proj)) {
+            return query.proj.some(function(key) {
+                return schema.iKeyMap[key] && schema.iKeyMap[key].type === 'static';
+            });
+        } else if (query.proj.constructor === String) {
+            return schema.iKeyMap[query.proj] && schema.iKeyMap[query.proj].type === 'static';
+        } else {
+            throw new Error('Unsupported query proj: ' + query.proj + ' of type ' + query.proj.constructor);
+        }
+    } else {
+        return (table === 'data') && Object.keys(schema.iKeyMap).some(function(key) {
+            return schema.iKeyMap[key].type === 'static';
+        });
+    }
+}
+
+dbu.staticTableExist = function(schema) {
+    return isStaticJoinNeeded(null, schema, 'data');
+};
+
+function constructLimit(query) {
+    var sql = '';
+    if (query.limit) {
+        sql += ' limit ' + query.limit;
+    }
+
+    if (query.next) {
+        sql += ' offset ' + query.next;
+    }
+    return sql;
+}
+
+dbu.buildGetQuery = function(keyspace, query, table, schema, includePreparedForDelete) {
+    var proj = constructProj(query, schema);
+    var order = constructOrder(query, schema);
+    var limit = constructLimit(query);
+    var params = [];
+    var condition = '';
+    var sql;
+
+    if (includePreparedForDelete === undefined) {
+        includePreparedForDelete = true;
+    }
+
+    if (query.attributes) {
+        var condResult = buildCondition(query.attributes, schema, includePreparedForDelete);
+        condition = ' where ' + condResult.query + ' ';
+        params = condResult.params;
+    }
+
+    if (query.index) {
+        var indexSchema = schema.secondaryIndexes[query.index];
+        if (!indexSchema) {
+            throw new Error('Index does not exist: ' + query.index);
+        }
```

Comment:
So basically, currently secondary indexes are ignored & we simply scan the entire table?